docs/docs/models-definition.md

```diff
@@ -256,92 +262,78 @@ the rule in a transaction. See [the transaction section](transactions/#options)
+    const parts = value.split(' ');
+    this.lastName = parts.pop();
+    this.firstName = parts.join(' ');
+  }
+}
+Employee.init({
+  firstName: {
+    type: Sequelize.STRING,
+    allowNull: false
   },
-  title: {
-    type     : Sequelize.STRING,
+  lastName: {
+    type: Sequelize.STRING,
     allowNull: false,
-    set      : function(val) {
-      this.setDataValue('title', val.toUpperCase());
-    }
   }
-});
+}, { sequelize });
 
 Employee
-  .create({ name: 'John Doe', title: 'senior engineer' })
-  .then(function(employee) {
-    console.log(employee.get('name')); // John Doe (SENIOR ENGINEER)
-    console.log(employee.get('title')); // SENIOR ENGINEER
+  .create({ firstName: 'John', lastName: 'Doe' })
+  .then(employee => {
+    console.log(employee.fullName); // John Doe
+    employee.fullName = 'Jack Bauer';
+    console.log(employee.firstName); // Jack
+    console.log(employee.lastName); // Bauer
   })
 ```
 
-### Defining as part of the model options
+### Defining as a VIRTUAL attribute
 
-Below is an example of defining the getters and setters in the model options. The `fullName` getter,  is an example of how you can define pseudo properties on your models - attributes which are not actually part of your database schema. In fact, pseudo properties can be defined in two ways: using model getters, or by using a column with the [`VIRTUAL` datatype](../api/datatypes#virtual). Virtual datatypes can have validations, while getters for virtual attributes cannot.
+Pseudo properties can also be defined by using a column with the [`VIRTUAL` datatype](../api/datatypes#virtual). Virtual datatypes can have validations, while getters for virtual attributes cannot.
 
 Note that the `this.firstname` and `this.lastname` references in the `fullName` getter function will trigger a call to the respective getter functions. If you do not want that then use the `getDataValue()` method to access the raw value (see below).
 
-```js
-var Foo = sequelize.define('foo', {
-  firstname: Sequelize.STRING,
-  lastname: Sequelize.STRING
-}, {
-  getterMethods   : {
-    fullName       : function()  { return this.firstname + ' ' + this.lastname }
-  },
 
-  setterMethods   : {
-    fullName       : function(value) {
-        var names = value.split(' ');
+### Bypassing Getters and setters
 
-        this.setDataValue('firstname', names.slice(0, -1).join(' '));
-        this.setDataValue('lastname', names.slice(-1).join(' '));
-    },
-  }
-});
-```
-
-### Helper functions for use inside getter and setter definitions
-
-* retrieving an underlying property value - always use `this.getDataValue()`
+If you want to define a getter or setter with the same name as your model attribute, you can prefix it and configure the `field` column option.
 
 ```js
-/* a getter for 'title' property */
-function() {
-    return this.getDataValue('title');
+class Transaction extends Sequelize.Model {
+  get status() {
+    return this._status.toLowerCase();
```

Comment:
Personally I dislike `getDataValue` because you cannot get any compile-time checks or IntelliSense for the attribute name, as it is a string.

But the primary motivation was that #6270 might change instances so that attributes are saved on the instance directly instead of in `dataValues` (and we only keep `previousDataValues`). There are a couple of reasons why this would be good, we wouldn't need getters and setters for the attributes anymore, which means better performance and that an instance can be treated like a plain simple instance of the model class. You can for example use the `delete` operator properly, which previously only deleted the getter/setter.

Prefixing the attribute with an underscore is the established pattern for implementing proxy accessors. Basically the actual property is private (and by convention prefixed with underscore) while the setters/getters are public.

For example http://stackoverflow.com/a/12850536/4208018 (it's TS but the same applies to ES6)